main/tasks/stratum_task.c

```diff
@@ -48,313 +44,335 @@ struct timeval tcp_rcv_timeout = {
 
 bool is_wifi_connected() {
     wifi_ap_record_t ap_info;
-    if (esp_wifi_sta_get_ap_info(&ap_info) == ESP_OK) {
-        return true;
-    } else {
-        return false;
-    }
-}
-
-void cleanQueue(GlobalState * GLOBAL_STATE) {
-    ESP_LOGI(TAG, "Clean Jobs: clearing queue");
-    GLOBAL_STATE->abandon_work = 1;
-    queue_clear(&GLOBAL_STATE->stratum_queue);
-
-    pthread_mutex_lock(&GLOBAL_STATE->valid_jobs_lock);
-    ASIC_jobs_queue_clear(&GLOBAL_STATE->ASIC_jobs_queue);
-    for (int i = 0; i < 128; i = i + 4) {
-        GLOBAL_STATE->valid_jobs[i] = 0;
-    }
-    pthread_mutex_unlock(&GLOBAL_STATE->valid_jobs_lock);
+    return esp_wifi_sta_get_ap_info(&ap_info) == ESP_OK;
 }
 
-void stratum_reset_uid(GlobalState * GLOBAL_STATE)
+void stratum_task_primary(void * pvParameters)
 {
-    ESP_LOGI(TAG, "Resetting stratum uid");
-    GLOBAL_STATE->send_uid = 1;
-}
+    GlobalState * GLOBAL_STATE = (GlobalState *) pvParameters;
 
+    StratumConnection * connection = &GLOBAL_STATE->connections[0];
 
-void stratum_close_connection(GlobalState * GLOBAL_STATE)
-{
-    if (GLOBAL_STATE->sock < 0) {
-        ESP_LOGE(TAG, "Socket already shutdown, not shutting down again..");
-        return;
-    }
+    stratum_task_init_connection(connection);
+    connection->id = 0;
+    connection->host = GLOBAL_STATE->SYSTEM_MODULE.pool_url;
+    connection->port = GLOBAL_STATE->SYSTEM_MODULE.pool_port;
+    connection->username = GLOBAL_STATE->SYSTEM_MODULE.pool_user;
+    connection->password = GLOBAL_STATE->SYSTEM_MODULE.pool_pass;
+
+    ESP_LOGI(TAG, "Opening connection to primary pool: %s:%d", connection->host, connection->port);
 
-    ESP_LOGE(TAG, "Shutting down socket and restarting...");
-    shutdown(GLOBAL_STATE->sock, SHUT_RDWR);
-    close(GLOBAL_STATE->sock);
-    cleanQueue(GLOBAL_STATE);
-    vTaskDelay(1000 / portTICK_PERIOD_MS);
+    stratum_task("primary_pool", GLOBAL_STATE, connection);
 }
 
-void stratum_primary_heartbeat(void * pvParameters)
+void stratum_task_secondary(void * pvParameters)
 {
     GlobalState * GLOBAL_STATE = (GlobalState *) pvParameters;
 
-    ESP_LOGI(TAG, "Starting heartbeat thread for primary pool: %s:%d", primary_stratum_url, primary_stratum_port);
-    vTaskDelay(10000 / portTICK_PERIOD_MS);
+    StratumConnection * connection = &GLOBAL_STATE->connections[1];
 
-    int addr_family = AF_INET;
-    int ip_protocol = IPPROTO_IP;
+    stratum_task_init_connection(connection);
+    connection->id = 1;
+    connection->host = GLOBAL_STATE->SYSTEM_MODULE.fallback_pool_url;
```

Comment:
For future consideration: move the pool settings into a struct, so stratum_task_primary and stratum_task_secondary can be merged and accept the id number and the pool settings struct. This continues moving towards n > 2 pool configs.

Reply:
Yep exactly. That is the intent here. I just kept it simple for the first iteration.